### PR TITLE
:zap: [RND-48076] Add dropdown button

### PR DESCRIPTION
**Pull request by Gladkowski**
- Preferred dropdown width is 280 px (verified by designers).

### CODE REVIEW

File: optimus/lib/src/dropdown_button.dart [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        return OptimusColors.neutral100;
+      case OptimusDropdownButtonType.primary:
+        return OptimusColors.primary700;
+      case OptimusDropdownButtonType.text:
+        return OptimusColors.neutral500t8;
+    }
+  }
+
+  // ignore: missing_return
+  Color get _highLightColor {
```
Comment (chaeMil) [inline optimus/lib/src/dropdown_button.dart:206]: Just a small thing, I would rename it to `_highlightColor` just lower the L

File: optimus/lib/src/dropdown_button.dart [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+            onExit: (_) => _onHoverChanged(false),
+            child: GestureDetector(
+              onTap: _node.requestFocus,
+              onTapDown: (_) => setState(() => _isTappedDown = true),
+              onTapUp: (_) => setState(() => _isTappedDown = false),
+              onTapCancel: () => setState(() => _isTappedDown = false),
+              child: Focus(
+                focusNode: _node,
+                child: Container(
+                  padding: const EdgeInsets.symmetric(horizontal: 16),
```
Comment (chaeMil) [inline optimus/lib/src/dropdown_button.dart:92]: Please use `spacing200` here

**Review by chaeMil — CHANGES_REQUESTED** [review]

**Review by chaeMil — APPROVED** [review]

File: optimus/lib/src/search/search_field_dropdown.dart [hunk 3]
```diff
@@ -50,15 +53,31 @@ class _OptimusSearchFieldDropdownState<T>
     final isOnTop = _topSpace > _bottomSpace;
     final maxHeight = max(_topSpace, _bottomSpace);
 
+    final isLeftToRight = _rightSpace >= widget.width + _widgetPadding;
```
Comment (ookami-kb) [inline optimus/lib/src/search/search_field_dropdown.dart:56]: Select is broken now.

**Review by ookami-kb — CHANGES_REQUESTED** [review]

**Comment by ookami-kb** [comment]
@Gladkowski @chaeMil I've pushed some updates. I hope I didn't break positioning, please check it.

**Comment by Gladkowski** [comment]
> @Gladkowski @chaeMil I've pushed some updates. I hope I didn't break positioning, please check it.

@ookami-kb @chaeMil _Right to Left_ dropdown position was broken. I've added little fix. Please check.

**Comment by chaeMil** [comment]
@ookami-kb @Gladkowski checked :)

**Review by ookami-kb — APPROVED** [review]

**Pull request merged**